File: AI-agent-frontend-vue3/src/api/notification.ts
// Copyright (c) 2025 左岚. All rights reserved.
/**
 * 消息通知 API
 * 对接 FastAPI 消息通知系统
 */
import { get, post, put, del } from './request'

export interface Notification {
  notification_id: number
  user_id: number
  title: string
  content: string
  type: string // system/test/error/info
  is_read: boolean
  create_time: string
  read_time?: string
}

export interface NotificationStats {
  total: number
  unread: number
  read: number
}

export interface NotificationListParams {
  filter_type?: string // all/unread/read/system/test/error/info
  skip?: number
  limit?: number
}

/**
 * 获取通知列表
 */
export function getNotificationList(params: NotificationListParams = {}) {
  return get<{ success: boolean; data: { items: Notification[]; total: number; page: number; page_size: number; pages: number } }>('/api/v1/notifications/', params)
}

/**
 * 获取通知统计
 */
export function getNotificationStats() {
  return get<{ success: boolean; data: NotificationStats }>('/api/v1/notifications/stats')
}

/**
 * 获取通知详情
 */
export function getNotificationDetail(notificationId: number) {
  return get<{ success: boolean; data: Notification }>(`/api/v1/notifications/${notificationId}`)
}

/**
 * 标记通知为已读
 */
export function markNotificationAsRead(notificationId: number) {
  return put<{ success: boolean; message: string; data: Notification }>(`/api/v1/notifications/${notificationId}/read`, {})
}

/**
 * 标记所有通知为已读
 */
export function markAllNotificationsAsRead() {
  return put<{ success: boolean; message: string; data: { count: number } }>('/api/v1/notifications/read-all', {})
}

/**
 * 删除通知
 */
export function deleteNotification(notificationId: number) {
  return del<{ success: boolean; message: string }>(`/api/v1/notifications/${notificationId}`)
}

/**
 * 删除所有通知
 */
export function deleteAllNotifications() {
  return del<{ success: boolean; message: string; data: { count: number } }>('/api/v1/notifications/')
}

File: AI-agent-backend-fastapi/app/schemas/notification.py
# Copyright (c) 2025 左岚. All rights reserved.
"""
消息通知Schema
"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class NotificationBase(BaseModel):
    """通知基础Schema"""
    title: str = Field(..., description="通知标题")
    content: str = Field(..., description="通知内容")
    type: str = Field(default="info", description="通知类型: system/test/error/info")


class NotificationCreate(NotificationBase):
    """创建通知Schema"""
    user_id: int = Field(..., description="接收用户ID")


class NotificationUpdate(BaseModel):
    """更新通知Schema"""
    is_read: Optional[bool] = Field(None, description="是否已读")


class NotificationResponse(NotificationBase):
    """通知响应Schema"""
    notification_id: int = Field(..., description="通知ID")
    user_id: int = Field(..., description="用户ID")
    is_read: bool = Field(..., description="是否已读")
    create_time: datetime = Field(..., description="创建时间")
    read_time: Optional[datetime] = Field(None, description="阅读时间")

    class Config:
        from_attributes = True


class NotificationListParams(BaseModel):
    """通知列表查询参数"""
    filter_type: Optional[str] = Field(None, description="过滤类型: all/unread/read/system/test/error/info")
    skip: int = Field(0, ge=0, description="跳过记录数")
    limit: int = Field(20, ge=1, le=100, description="返回记录数")


class NotificationStats(BaseModel):
    """通知统计Schema"""
    total: int = Field(..., description="总数")
    unread: int = Field(..., description="未读数")
    read: int = Field(..., description="已读数")

